### ISSUE TITLE
cmake -DCMAKE issue.

### ISSUE THREAD

**Issue by blackranger07** [ripple/validator-keys-tool#28]
During the cmake step, I get the following output. I even tried adding the path to rippled but still doesn't work. I know I'm missing something.

root@parallels-vm:/home/parallels/VALIDATOR-TOOL/build/gcc.debug# cmake -DCMAKE_BUILD_TYPE=Release
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.

CMake Error: The source directory "/home/parallels/VALIDATOR-TOOL/build/gcc.debug" does not appear to contain CMakeLists.txt.

**Comment by legleux** [comment]
Hi, hope you figured this out already.
You missed the `../..` in the cmake command.
It's `cmake -DCMAKE_BUILD_TYPE=Release ../..`
That is telling cmake the path to the source files (where the CMakeLists.txt file lives.)

**Comment by blackranger07** [comment]
Hi legleux,

Yeah I made sure to include previously and I still get the error. Here are the steps that were taken. I appreciate the help.

**Output from install...**
```
CMake Error: The source directory "/home/parallels/VALIDATOR-TOOL" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
Error: could not load cache
./validator-keys: No such file or directory
```

 **Steps taken**
```
mkdir VALIDATOR-TOOL
  DIR=`pwd`
  cd ${DIR}/VALIDATOR-TOOL
  mkdir -p build/gcc.debug && cd build/gcc.debug
  apt install cmake -y
  git clone https://github.com/ripple/validator-keys-tool.git
  cmake -DCMAKE_BUILD_TYPE=Release ../..
  cmake --build .
  ./validator-keys
```

**Comment by blackranger07** [comment]
I think I see where I may have gone wrong.

**Comment by legleux** [comment]
Ignore if you aleady saw it but...
`../..` is the directory where the source code lives, the root `CMakeLists.txt`.
You created the build directory then `cd`'d to it _then_ cloned the repo in _that_ , then called cmake on a directory 2 dirs up (your starting dir, where there is no CMakeLists.txt, hence the error.)
Your post must be a script bc you would have gotten the error you posted after the  `cmake -DCMAKE_BUILD_TYPE=Release ../..` call.
Step 1. Cmake configure `cmake -S <source_dir> -B <build_dir>`
Step 2. Build it `cmake --build <build_dir>`

**Comment by intelliot** [comment]
Sounds like this has been resolved, but if not, feel free to open a new issue. Thanks!
